impl/src/main/java/jakarta/faces/model/SelectItemGroup.java

```diff
@@ -168,7 +168,7 @@ public void setSelectItems(Collection<? extends SelectItem> selectItems) {
         if (selectItems == null) {
             throw new NullPointerException();
         }
-        setSelectItems(selectItems.toArray(new SelectItem[selectItems.size()]));
+        setSelectItems(selectItems.toArray(new SelectItem[0]));
```

Comment:
This will now create an empty array for nothing. This is only necessary when you can't guarantee that `selectItems` is immutable at the moment you want to convert to array.